zulipterminal/model.py

```diff
@@ -296,32 +296,9 @@ def _update_initial_data(self) -> None:
                                                 num_after=10,
                                                 num_before=30,
                                                 anchor=None),
+                'register': executor.submit(self._register_desired_events,
+                                            fetch_data=True),
             }  # Dict[str, Future[Any]]
-            try:
-                response = self.client.register(
-                    event_types=Model.event_types,
-                    fetch_event_types=[
-                        'presence',
-                        'subscription',
-                        'message',
-                        'update_message_flags',
-                        'muted_topics',
-                        'realm_user',  # Enables cross_realm_bots
-                    ],
-                    client_gravatar=True,
-                    apply_markdown=True,
-                )
-                self.initial_data.update(response)
-                self.max_message_id = response['max_message_id']
-                self.queue_id = response['queue_id']
-                self.last_event_id = response['last_event_id']
-            except zulip.ZulipError as e:
-                raise ServerConnectionFailure(e)
-            except Exception as e:
-                # There was some error in the request sent.
-                if response['result'] == 'error':
```

Comment:
Are we considering this case where an error occurs in the request?

Reply:
The code now explicitly checks for the inverse, `response['result'] == 'success'` instead; we don't get the explicit type of error now, ie. there is now no handling of general exceptions.

Regarding other exceptions:
 * We could catch other exceptions that we understand from the `register` call, but it should just be `ZulipError`? If there are others to handle, we'd want to distinguish them.
 * This code is now all called in extra threads (though was already before, in one case!), so we would (should!) have to pass *unhandled* general exceptions/errors back cleanly into the main thread for logging/shutdown as we do with exceptions in the main thread. That's a bit more complex than this PR was intended to be.

As far as I'm aware we've not had any exceptions logged (and reported) via the code that was removed except the bug I mentioned - so my feeling would be to explore these two cases separately.

Reply:
All-right! This definitely clears things. Thanks for the detailed explanation.